go-controller/pkg/crd/clusternetworkconnect/v1/types.go

```diff
@@ -0,0 +1,188 @@
+
+	// connectSubnets specifies the subnets used for interconnecting the selected networks.
+	// This creates a shared subnet space that connected networks can use to communicate.
+	// Can have at most 1 CIDR for each IP family (IPv4 and IPv6).
+	// Must not overlap with:
+	//  any of the pod subnets used by the selected networks.
+	//  any of the transit subnets used by the selected networks.
+	//  any of the service CIDR range used in the cluster.
+	//  any of the join subnet of the selected networks to be connected.
+	//  any of the masquerade subnet range used in the cluster.
+	//  any of the node subnets chosen by the platform.
+	//  any of other connect subnets for other ClusterNetworkConnects that might be selecting same networks.
+	//
+	// Does not have a default value for the above reason so
+	// that user takes care in setting non-overlapping subnets.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="connectSubnets is immutable"
+	// +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isCIDR(self[0].cidr) || !isCIDR(self[1].cidr) || cidr(self[0].cidr).ip().family() != cidr(self[1].cidr).ip().family()", message="When 2 CIDRs are set, they must be from different IP families"
+	ConnectSubnets []ConnectSubnet `json:"connectSubnets"`
+
+	// connectivityEnabled specifies which connectivity types should be enabled for the connected networks.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="connectivityEnabled cannot contain duplicate values"
+	ConnectivityEnabled []ConnectivityType `json:"connectivityEnabled"`
+}
+
+// +kubebuilder:validation:XValidation:rule="isCIDR(self) && cidr(self) == cidr(self).masked()", message="CIDR must be a valid network address"
+// +kubebuilder:validation:MaxLength=43
+type CIDR string
+
+// +kubebuilder:validation:XValidation:rule="!has(self.networkPrefix) || !isCIDR(self.cidr) || self.networkPrefix > cidr(self.cidr).prefixLength()", message="NetworkPrefix must be smaller than CIDR subnet"
+// +kubebuilder:validation:XValidation:rule="!has(self.networkPrefix) || !isCIDR(self.cidr) || (cidr(self.cidr).ip().family() != 4 || self.networkPrefix < 32)", message="NetworkPrefix must < 32 for ipv4 CIDR"
+type ConnectSubnet struct {
+	// CIDR specifies ConnectSubnet, which is split into smaller subnets for every connected network.
+	// This CIDR should be containing 2*((Number of L3 networks*Max Number of Nodes)+Number of L2 networks) IPs.
+	// Example: cidr= "192.168.0.0/16", networkPrefix= 24 means that you can connect 256 Layer3 networks
+	// if the cluster has 128 nodes and 0 layer2 networks OR 255 Layer3 networks and 128 Layer2 networks
```

Comment:
is this sentence a bit broken? should it be:
`if the cluster has 128 nodes you can connect 256 Layer3 networks and 0 layer2 networks OR 255 Layer3 networks and 128 Layer2 networks?`